grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1793,15 +1796,15 @@ class StockMovementService {
 
     void createOrUpdatePicklistItem(StockMovementItem stockMovementItem, PicklistItem picklistItem,
                                     InventoryItem inventoryItem, Location binLocation,
-                                    Integer quantity, String reasonCode, String comment) {
+                                    Integer quantity, String reasonCode, PickType pickType, String comment) {
 
         RequisitionItem requisitionItem = RequisitionItem.get(stockMovementItem.id)
-        createOrUpdatePicklistItem(requisitionItem, picklistItem, inventoryItem, binLocation, quantity, reasonCode, comment)
+        createOrUpdatePicklistItem(requisitionItem, picklistItem, inventoryItem, binLocation, quantity, reasonCode, pickType, comment)
     }
 
     void createOrUpdatePicklistItem(RequisitionItem requisitionItem, PicklistItem picklistItem,
                                     InventoryItem inventoryItem, Location binLocation,
-                                    Integer quantity, String reasonCode, String comment) {
+                                    Integer quantity, String reasonCode, PickType pickType, String comment) {
```

Comment:
this method is getting overloaded more and more, but let's say I don't look at it until we refactor this feature completely :see_no_evil:

Reply:
yeah stock movement is definitely in need of a refactor but that's going to need to get done as a dedicated task because it's so massive... not looking forward to that

Reply:
@EWaterman @awalkowiak Evergreen comment. This gets mentioned every few weeks, but it's such a big refactoring and we've had other big tech debt issues that have been priorities that we just keep kicking the can. But we really should schedule some time soon to do tech debt backlog grooming to understand the different refactorings involved and identify some discoveries that are worth taking on in one of the next release or two.